### ISSUE TITLE
Documenting our new code review process

### ISSUE THREAD

**Issue by itsrachelfish** [OpenKore/openkore#9]
@allanon and I have been discussing changes we'd like to see in regards to the way that code is submitted into the project. When we were using SVN, anyone with commit access would write code and push it up whenever they felt like it. There was rarely any communication between developers before changes were made or even testing done on different platforms / servers.

We'd like to make the entire development process more reliable and transparent by utilizing pull requests here on GitHub. So far we've already started using it amongst ourselves, but official documentation should be written to replace the old SVN guides like these:

http://wiki.openkore.com/index.php/Patches_Welcome
http://wiki.openkore.com/index.php/Development_Conventions

**tldr; Don't push to master.** Make a pull request and have it approved by another developer first.

**Comment by windhamwong** [comment]
I love the idea but pushing to branch may affect the updates for servers.
The reviewing process is good only when all of our developers are active however we all know that we are so busy on other things and rarely updating or reviewing the codes and the system.

**Comment by windhamwong** [comment]
nice. I will join irc later today.
I will just create a twRO branch for updating changes here.

**Comment by parasquid** [comment]
I don't think it's a good idea to "maintain" a server branch.

Each branch created should be a feature branch. For example, if you need to updated twRO fld files, you'll just create a branch called "2016-03-03-updated-twro-fld-files" and create a PR against that.

It's the Git :tm: way :smile:

When I first started using Git from SVN I also had to adapt my workflow, and now that I'm more used to a distributed version control system. it now feels natural. Unlike SVN, branches in Git are very cheap and very fast to make and we should take advantage of the workflow it provides.

**Comment by lututui** [comment]
Redirected "Current Development" link to pull requests

http://openkore.com/index.php?title=Template:Index/MainLinks&curid=963&diff=3698&oldid=3687
